src/renderer/k8s/api-base.injectable.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { apiBaseInjectionToken } from "../../common/k8s-api/api-base";
+import createJsonApiInjectable from "../../common/k8s-api/create-json-api.injectable";
+import { apiPrefix, isDebugging } from "../../common/vars";
+import isDevelopmentInjectable from "../../common/vars/is-development.injectable";
+
+const apiBaseInjectable = getInjectable({
+  id: "api-base",
+  instantiate: (di) => {
+    const createJsonApi = di.inject(createJsonApiInjectable);
+
+    return createJsonApi(
+      {
+        serverAddress: `http://127.0.0.1:${window.location.port}`,
+        apiBase: apiPrefix,
+        debug: di.inject(isDevelopmentInjectable) || isDebugging,
```

Comment:
Hmm. Extract stuff like `isDebugging`, `window.location.host`, `window.location.port` and `apiPrefix` as injectables with side-effect.

Reply:
`apiPrefix` is not a side effect but will handle the others.